### PR TITLE
Update/domains plans changes

### PR DESCRIPTION
**Pull request by rralian**
#### Changes proposed in this Pull Request

This PR updates several different warnings, nudges, and general communication about the domain credit offered with a paid plan. Domain credits will be offered with the initial purchase of a plan, but renewals will be charged at their regular price.

#### Testing instructions

Honestly, it'd be best to just have someone who knows how to trigger these give it a look. :-) Other than the linting issue I address (separate commit), there are no logic changes to test, everything is text. So I think it's fine to just have @semihakocer and/or @anneforbush give the content a look.

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx [hunk 1]
```diff
@@ -34,17 +34,18 @@ const CustomDomainPurchaseDetail = ( {
 				description={
 					onlyBlogDomain
 						? translate(
-								'Your plan includes a free .blog domain, which gives your site a more professional, branded feel.'
+								'Your plan includes a free .blog domain for one year, which gives your site a more professional, branded feel.'
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx]: :ok: This change is safe, we'll keep the existing 17 translations.

File: client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx [hunk 2]
```diff
@@ -34,17 +34,18 @@ const CustomDomainPurchaseDetail = ( {
...
 						  )
 						: translate(
-								'Your plan includes a free custom domain, which gives your site a more professional, branded feel.'
+								'Your plan includes a free custom domain for one year, which gives your site a more professional, branded feel.'
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx]: :ok: This change is safe, we'll keep the existing 20 translations.

File: client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx [hunk 3]
```diff
@@ -53,7 +54,7 @@ const CustomDomainPurchaseDetail = ( {
 				icon={ <img alt="" src={ customDomainIcon } /> }
 				title={ translate( 'Custom Domain' ) }
 				description={ translate(
-					'Your plan includes the custom domain {{em}}%(siteDomain)s{{/em}}, your own personal corner of the web.',
+					'Your plan includes one year of your custom domain {{em}}%(siteDomain)s{{/em}}, your own personal corner of the web.',
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx]: :ok: This change is safe, we'll keep the existing 19 translations.

File: client/my-sites/checkout/checkout/free-cart-payment-box.jsx [hunk 4]
```diff
@@ -42,7 +42,7 @@ class FreeCartPaymentBox extends React.Component {
 							<span>
 								{ cart.has_bundle_credit
 									? this.props.translate(
-											'You get one free domain with your subscription to %(productName)s. Time to celebrate!',
+											'You get a free domain for one year with your subscription to %(productName)s. Time to celebrate!',
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout/free-cart-payment-box.jsx]: :ok: This change is safe, we'll keep the existing 25 translations.

File: client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx (hunk 1 above)
Comment (rralian) [reply client/my-sites/checkout/checkout-thank-you/custom-domain-purchase-detail.jsx]: Not sure what this means. The text change is meant to clarify policy so I would expect new translations to be necessary. /cc @akirk

File: client/lib/plans/constants.js [hunk 5]
```diff
@@ -1279,7 +1279,7 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_BLOG_DOMAIN ]: {
 		getSlug: () => FEATURE_BLOG_DOMAIN,
-		getTitle: () => i18n.translate( 'Custom .blog Domain' ),
+		getTitle: () => i18n.translate( 'Free .blog Domain for One Year' ),
```
Comment (a8ci18n) [inline client/lib/plans/constants.js]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Free .blog domain for one year' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (rralian) [reply client/lib/plans/constants.js]: Capitalization is important here.

File: client/lib/plans/constants.js [hunk 6]
```diff
@@ -1119,19 +1119,19 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_FREE_BLOG_DOMAIN ]: {
 		getSlug: () => FEATURE_ADVANCED_CUSTOMIZATION,
-		getTitle: () => i18n.translate( 'Free .blog domain' ),
+		getTitle: () => i18n.translate( 'Free .blog Domain for One Year' ),
```
Comment (a8ci18n) [inline client/lib/plans/constants.js]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Free .blog domain for one year' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/lib/plans/constants.js [hunk 7]
```diff
@@ -1119,7 +1119,7 @@ export const FEATURES_LIST = {
 
 	[ FEATURE_FREE_BLOG_DOMAIN ]: {
 		getSlug: () => FEATURE_ADVANCED_CUSTOMIZATION,
-		getTitle: () => i18n.translate( 'Free .blog domain for one year' ),
+		getTitle: () => i18n.translate( 'Free .blog Domain for One Year' ),
```
Comment (rralian) [inline client/lib/plans/constants.js]: I'm not certain where or even whether this string is being used, but there are several features whose titles are not capitalized. The capitalized ones are used in the plans pages (both /plans and signup). I've left the capitalization in place and just updated the content for policy change.

File: client/lib/plans/constants.js [hunk 8]
```diff
@@ -1288,15 +1291,17 @@ export const FEATURES_LIST = {
...
+				'Get a free .blog domain for one year. Premium domains not included. Your domain will renew at its regular price.'
 			);
 		},
 	},
 
 	[ FEATURE_CUSTOM_DOMAIN ]: {
 		getSlug: () => FEATURE_CUSTOM_DOMAIN,
-		getTitle: () => i18n.translate( 'Custom Domain Name' ),
+		getTitle: () =>
+			i18n.translate( 'Free Domain for One Year', {
```
Comment (a8ci18n) [inline client/lib/plans/constants.js]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'Free Domain for One Year' )` *ES Score: 12*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

**Comment by semihakocer** [comment]
Thanks for this PR! I have checked the copy, all looks good.

**Pull request merged**